pykafka/balancedconsumer.py

```diff
@@ -235,8 +239,17 @@ def stop(self):
         This method should be called as part of a graceful shutdown process.
         """
         self._zookeeper.stop()
-        self._consumer.stop()
+        # If internal consumer is not running and this consumer is running,
```

Comment:
Final question (I hope) and not actually new in this pullreq:
shouldn't `_zookeeper.stop` actually come last here? As long as `_consumer` still runs it might commit offsets, so it should keep exclusive hold of its partitions, right?